### PR TITLE
Add controller logic for Debugger Variables view

### PR DESCRIPTION
**Pull request by kenzieschmoll**
The majority of this code was ported directly from the html debugger variables view. Actual display for this data to come in a follow up PR.

### CODE REVIEW

File: packages/devtools_app/lib/src/debugger/flutter/debugger_model.dart [hunk 1]
```diff
@@ -182,3 +184,64 @@ class StackFrameAndSourcePosition {
...
+      }
+    } else if (value is Sentinel) {
+      valueStr = value.valueAsString;
+    } else if (value is TypeArgumentsRef) {
+      valueStr = value.name;
+    } else {
+      valueStr = value.toString();
+    }
+
+    return '$name: $valueStr';
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_model.dart:237]: When we do implement the `Variables` widget (below), we can use some style differences to call out the variable name and value differently - we won't have to just concatenate them w/ a `:` separator.

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart [hunk 2]
```diff
@@ -458,15 +462,81 @@ class DebuggerController extends DisposableController
...
+    }
+
     if (frame?.scriptRef != null) {
       showScriptLocation(
           ScriptLocation(frame.scriptRef, location: frame.sourcePosition));
     }
   }
 
+  Future<List<Variable>> _variablesForFrame(Frame frame) async {
+    final vars = frame.vars.map((v) => Variable.create(v)).toList();
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:479]: `vars` ==> `variables`?

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart [hunk 3]
```diff
@@ -458,15 +462,81 @@ class DebuggerController extends DisposableController
...
+
+    for (var v in vars) {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:481]: ==> `final variable`?

File: packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart [hunk 4]
```diff
@@ -458,15 +462,81 @@ class DebuggerController extends DisposableController
...
+      _variables.value = await _variablesForFrame(frame.frame);
+    }
+
     if (frame?.scriptRef != null) {
       showScriptLocation(
           ScriptLocation(frame.scriptRef, location: frame.sourcePosition));
     }
   }
 
+  Future<List<Variable>> _variablesForFrame(Frame frame) async {
```
Comment (devoncarew) [inline packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart:478]: This method is longer, w/ a few levels of nesting. Can this be broken out into a few smaller helper methods? Or, if we expect more complexity here later, perhaps into a helper class of some kind?

**Comment by kenzieschmoll** [comment]
closing in favor of https://github.com/flutter/devtools/pull/1909

**Pull request closed without merging**